labrad/types.py

```diff
@@ -707,13 +725,13 @@ def __unflatten__(self, s, endianness):
         return s.get(n)
 
     def __flatten__(self, s, endianness):
-        if not isinstance(s, str):
+        if not isinstance(s, bytes):
             raise FlatteningError(s, self)
         return struct.pack(endianness + 'I', len(s)) + s, self
 
-# Since bytes is an alias for str in python 2.7, don't enable this
-# until we are ready to cut over.
-# registerType(bytes, LRBytes)
+# bytes is an alias for str in python 2.7, but in python 3 they are different.
+if bytes != str:
+    registerType(bytes, LRBytes())
```

Comment:
The official [porting guide](https://docs.python.org/3/howto/pyporting.html#use-feature-detection-instead-of-version-detection) suggests using feature detection rather than version detection, which is why I did it this way. I'm open to changing it though if you feel like this is unclear.